schema/db_migration.sql

```diff
@@ -0,0 +1,212 @@
+--   Should ENUM instead of smallint for better performance (don't need to convert int to value later).
+--   Consider to add "created_at" column (maybe index for it also) in this table to track when a new
+--     record is created.
+--   Consider to add unique (might be partial) index on remainder to make sure no duplication txs,
+--      at least for txhash, contract...
+CREATE TYPE id_type AS ENUM ('eoa', 'contract', 'txhash', 'blockhash');
+CREATE TABLE ids (
+   id bigserial PRIMARY KEY, -- idx index created automatically for primary key (ids_pkey)
+   remainder bytea NOT NULL,
+   id_type id_type NOT NULL,
+   created_at int NOT NULL
+);
+
+-- nicks:
+--  id: bigint # foreign key id@ids
+--  nick: string # nickname of the particular id
+--  type: smallint # indicates nickname type
+-- TAMCHAU:
+--   String type should be converted to "varchar" with maximum is 255 chars.
+--   Changes the name of column to "nick_type".
+--   If the type is limited, we should consider to use `enum` as above.
+--   According to the filter function later, we should add index for nick or nick_type if needed.
+CREATE TABLE nicks (
+    id bigint NOT NULL,
+    nick varchar(255) NOT NULL,
+    nick_type smallint NOT NULL,
+    CONSTRAINT ids_nick_fk
+        FOREIGN KEY(id)
+	    REFERENCES ids(id)
+);
+
+-- erc20tokens:
+--  id: bigint # foreign key id@ids
+--  name: string # token name
+--  symbol: string # token symbol
+--  decimals: smallint # how many decimals the token has
+--  supply: bigint # divided by decimals
+-- TAMCHAU:
+--   "name" is internal key -> "token_name".
+--    Symbol should have maximum 100 chars.
+--    According to the filter function later, we should add index for symbol if needed.
+
+CREATE TABLE erc20tokens (
+    id bigint NOT NULL,
+    token_name varchar(255) NOT NULL,
+    symbol varchar(100) NOT NULL,
+    decimals smallint NOT NULL,
+    supply bigint NOT NULL,
+    CONSTRAINT ids_erc20token_fk
+        FOREIGN KEY(id)
+	    REFERENCES ids(id)
+);
+
+-- blocks:
+--   id: bigint # foreign key id@ids, the blockhash
+--   height: int # block height in the chain
+--   created_at: int # unix ts
+-- TAMCHAU:
+--   We can get rid "created_at" column here since it's present in ids table.
+--   "height" column is referenced from txs table, so we should add index for it.
+CREATE TABLE blocks (
+    id bigint NOT NULL,
+    height int NOT NULL,
+    CONSTRAINT ids_block_fk
+        FOREIGN KEY(id)
+	    REFERENCES ids(id)
+);
+CREATE INDEX index_blocks_on_height ON blocks (height); -- Should be UNIQUE?
+
+-- txs:
+--  txhash: bigint # foreign key id@ids, the txhash
+--  block: int # block height, foreign key height@blocks
+--  value: bigint # in gwei
+--  from_id: bigint # foreign key id@ids
+--  to_id: bigint # foreign key id@ids
+--  gas_limit: bigint
+--  gas_price: bigint
+--  method_id: int? # 4 bytes
+--  params: bytea? # not contract deployment, not erc20
+-- TAMCHAU:
+--  "txhash" is changed to "txhash_id" for more clear.
```

Comment:
ok, nice improvement, still it's better to create separated PR for this so we all are clear about what is original implementation, and what is improvement later